modules/nw-networkpolicy-deny-all-egress-network-policy.adoc

```diff
@@ -0,0 +1,101 @@
+
+[WARNING]
+====
+Without configuring a `NetworkPolicy` custom resource (CR) that allows traffic communication, the following policy might cause communication problems across your cluster.
+Creating a `default-deny-all` network policy that isolates pods for egress breaks pod-to-pod communication for ingress. This behavior is expected, and every connection allowed in the ingress direction must also allowlist connections in the egress direction. After creating a `default-deny-all` network policy, you should "Configure internet for egress pods" and "Enable pod-to-pod communication". 
+====
+
+.Prerequisites
+
+* Your cluster uses a network plugin that supports `NetworkPolicy` objects, such as the OVN-Kubernetes network plugin, with `mode: NetworkPolicy` set.
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with admin privileges.
+* You have configured an ingress network policy.
+* You have created at least two projects in your cluster.
+* You have created pods in your cluster.
+
+.Procedure
+
+. Create the following YAML that defines a `default-deny-all-egress` network policy to deny egress for all pods in the namespace. Save the YAML in the `default-deny-all-egress.yaml` file:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all-egress
+spec:
+  podSelector: {}
+  egress:
+  - to:
+    ports:
+    - protocol: TCP
+      port: 53 <1>
+    - protocol: UDP
+      port: 53 <1>
+  policyTypes:
+  - Egress
+----
+<1> Allows connections to port `53` on any IP to facilitate DNS lookups. 
+
+. Apply the policy by entering the following command:
++
+[IMPORTANT]
+====
+Do not apply this network policy to the `kube-system` namespace, as it can break cluster functionality.
+====
++
+[source,terminal]
+----
+$ oc apply -f default-deny-all-egress.yaml -n project-a
+----
+
+. Because network policies are namespaced resources, you must create this network policy for each namespace. Apply the policy to other applicable namespaces by entering the following command:
++
+[source,terminal]
+----
+$ oc apply -f default-deny-all-egress.yaml -n project-b
+----
++
+With the application of the `default-deny-all-egress` network policy, pods in those namespaces cannot receive external traffic.
+
+.Verification
+
+. Test egress connection on a pod by entering the following command. If the network policy was successfully applied, then the pod is unable to reach the endpoint: 
++
+[source,terminal]
+----
+$ oc exec -it test-pod-b -n project-b -- nslookup google.com
+----
++
+.Example output
++
+[source,terminal]
+----
+;; connection timed out; no servers could be reached
+----
+
+. Test ingress connection between pods in the `project-a` and `project-b` namespaces by entering the following command. Because the `default-deny-all-egress` network policy breaks pod-to-pod communication for ingress, pods should not longer be able to communicate.
```

Comment:
```suggestion
. Test ingress connection between pods in the `project-a` and `project-b` namespaces by entering the following command. Because the `default-deny-all-egress` network policy breaks pod-to-pod communication for egress, pods should not longer be able to communicate.
```